apps/sms/views/shared/js/startup.js

```diff
@@ -94,7 +97,20 @@ var Startup = exports.Startup = {
       // Init UI Managers
       TimeHeaders.init();
       ConversationView.init();
-      MessagingClient.init(App.instanceId);
+
+      // Fetch mmsSizeLimitation and max concat
+      Settings.init();
+
+      var serviceEndpoint;
+      const servicePath = '/services/js/messaging/messaging_service.js';
+      if (Settings.useSharedWorkerServices) {
+        serviceEndpoint = new SharedWorker(servicePath);
+      } else {
+        serviceEndpoint = exports;
+        LazyLoader.load(servicePath).then(() => MessagingService.init());
```

Comment:
Yeah, I think it was about +15ms for fullyLoaded - could be a margin of error though, but I totally agree I'd not do it for 1mb win, or maybe just switch v2.5 totally to window context service without any ifs